patcher/src/ts/sidebar/CharacterSelect.tsx

```diff
@@ -87,17 +97,35 @@ class CharacterSelect extends React.Component<CharacterSelectProps, CharacterSel
   }
 
   getSelectedIndex = () : number => {
-    return this.props.characters.findIndex((c: restAPI.SimpleCharacter) => c.id === this.props.selectedCharacter.id);
+    let selectedIndex = this.characters.findIndex((c: restAPI.SimpleCharacter) => c.id === this.selectedCharacter.id);
+    return selectedIndex;
+  }
+
+  onSelectedCharacterChanged = (character: restAPI.SimpleCharacter) => {
+    const {dispatch} = this.props;
+    dispatch(selectCharacter(character));
+    events.fire('play-sound', 'select');
   }
 
   render() {
-    if (this.props.characters.length == 0) {
+    const {charactersState, serversState} = this.props;
+
+    this.characters = charactersState.characters.filter((c: restAPI.SimpleCharacter) => c.shardID == serversState.currentServer.shardID || c.shardID == 0);
+    this.selectedCharacter = null;
+
+    if (!this.characters || this.characters.length == 0) {
       return (
         <div className="character-select-none">CREATE CHARACTER</div>
       );
     }
+
+    if(charactersState.selectedCharacter && charactersState.selectedCharacter.shardID == serversState.currentServer.shardID)
```

Comment:
https://github.com/airbnb/javascript#blocks--braces

    if (expression) statement;

or

    if (expression) {
      statement;
    }

are the acceptable forms.